Unable to build on linux.

Linux xubyruby 5.11.0-40-generic 
#44~20.04.2-Ubuntu SMP 
x86_64 x86_64 x86_64 GNU/Linux
_____________________________________________________________________________
I Get This:
_____________________________________________________________________________

Scanning dependencies of target translations
Scanning dependencies of target lua
[  1%] mo-update [zh_TW]: Creating locale directory.
[  1%] mo-update [be]: Creating locale directory.
[  2%] mo-update [bg]: Creating locale directory.
[  2%] Building C object lib/lua/build/CMakeFiles/lua.dir/lapi.c.o
[  2%] mo-update [ca]: Creating locale directory.
[  2%] mo-update [cs]: Creating locale directory.
[  2%] mo-update [da]: Creating locale directory.
[  2%] mo-update [de]: Creating locale directory.
[  3%] mo-update [el]: Creating locale directory.
[  3%] mo-update [eo]: Creating locale directory.
[  3%] mo-update [es]: Creating locale directory.
[  3%] mo-update [et]: Creating locale directory.
[  3%] mo-update [eu]: Creating locale directory.
[  4%] mo-update [fi]: Creating locale directory.
[  4%] mo-update [fil]: Creating locale directory.
[  4%] mo-update [fr]: Creating locale directory.
[  4%] mo-update [gd]: Creating locale directory.
[  4%] mo-update [gl]: Creating locale directory.
[  5%] mo-update [hu]: Creating locale directory.
[  5%] Building C object lib/lua/build/CMakeFiles/lua.dir/lauxlib.c.o
[  5%] mo-update [id]: Creating locale directory.
[  5%] mo-update [it]: Creating locale directory.
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/lauxlib.c: In function ‘luaL_loadfile’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/lauxlib.c:577:4: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
  577 |    while ((c = getc(lf.f)) != EOF && c != LUA_SIGNATURE[0]) ;
      |    ^~~~~
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/lauxlib.c:578:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘while’
  578 |     lf.extraline = 0;
      |     ^~
[  5%] mo-update [ja]: Creating locale directory.
[  5%] mo-update [jbo]: Creating locale directory.
[  5%] mo-update [kk]: Creating locale directory.
[  6%] mo-update [ko]: Creating locale directory.
[  6%] mo-update [ky]: Creating locale directory.
[  6%] mo-update [lt]: Creating locale directory.
[  6%] mo-update [lv]: Creating locale directory.
[  6%] mo-update [mr]: Creating locale directory.
[  6%] Building C object lib/lua/build/CMakeFiles/lua.dir/lbaselib.c.o
[  7%] mo-update [ms]: Creating locale directory.
[  7%] mo-update [nb]: Creating locale directory.
[  7%] mo-update [nl]: Creating locale directory.
[  7%] mo-update [nn]: Creating locale directory.
[  7%] mo-update [pl]: Creating locale directory.
[  8%] mo-update [pt]: Creating locale directory.
[  8%] mo-update [pt_BR]: Creating locale directory.
[  8%] mo-update [ro]: Creating locale directory.
[  8%] mo-update [ru]: Creating locale directory.
[  9%] Building C object lib/lua/build/CMakeFiles/lua.dir/lcode.c.o
[  9%] mo-update [sk]: Creating locale directory.
[ 10%] mo-update [sl]: Creating locale directory.
[ 10%] mo-update [sr_Cyrl]: Creating locale directory.
[ 10%] mo-update [sr_Latn]: Creating locale directory.
[ 10%] mo-update [sv]: Creating locale directory.
[ 10%] mo-update [sw]: Creating locale directory.
[ 11%] mo-update [tr]: Creating locale directory.
[ 11%] mo-update [tt]: Creating locale directory.
[ 11%] mo-update [uk]: Creating locale directory.
[ 11%] mo-update [vi]: Creating locale directory.
[ 11%] mo-update [zh_CN]: Creating locale directory.
[ 11%] mo-update [zh_TW]: Creating mo file.
[ 11%] Building C object lib/lua/build/CMakeFiles/lua.dir/ldblib.c.o
[ 11%] mo-update [be]: Creating mo file.
[ 11%] mo-update [bg]: Creating mo file.
[ 12%] mo-update [ca]: Creating mo file.
[ 12%] mo-update [cs]: Creating mo file.
[ 12%] Building C object lib/lua/build/CMakeFiles/lua.dir/ldebug.c.o
[ 12%] mo-update [da]: Creating mo file.
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ldebug.c: In function ‘symbexec’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ldebug.c:268:22: warning: this statement may fall through [-Wimplicit-fallthrough=]
  268 | #define check(x)  if (!(x)) return 0;
      |                      ^
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ldebug.c:272:26: note: in expansion of macro ‘check’
  272 | #define checkreg(pt,reg) check((reg) < (pt)->maxstacksize)
      |                          ^~~~~
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ldebug.c:414:9: note: in expansion of macro ‘checkreg’
  414 |         checkreg(pt, a+3);
      |         ^~~~~~~~
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ldebug.c:416:7: note: here
  416 |       case OP_JMP: {
      |       ^~~~
[ 12%] mo-update [de]: Creating mo file.
[ 12%] mo-update [el]: Creating mo file.
[ 13%] mo-update [eo]: Creating mo file.
[ 13%] mo-update [es]: Creating mo file.
[ 13%] mo-update [et]: Creating mo file.
[ 13%] mo-update [eu]: Creating mo file.
[ 13%] Building C object lib/lua/build/CMakeFiles/lua.dir/ldo.c.o
[ 13%] mo-update [fi]: Creating mo file.
[ 13%] mo-update [fil]: Creating mo file.
[ 14%] mo-update [fr]: Creating mo file.
[ 14%] mo-update [gd]: Creating mo file.
[ 14%] mo-update [gl]: Creating mo file.
[ 14%] mo-update [hu]: Creating mo file.
[ 14%] Building C object lib/lua/build/CMakeFiles/lua.dir/ldump.c.o
[ 14%] mo-update [id]: Creating mo file.
[ 15%] mo-update [it]: Creating mo file.
[ 16%] Building C object lib/lua/build/CMakeFiles/lua.dir/lfunc.c.o
[ 16%] mo-update [ja]: Creating mo file.
[ 16%] mo-update [jbo]: Creating mo file.
[ 16%] Building C object lib/lua/build/CMakeFiles/lua.dir/lgc.c.o
[ 16%] mo-update [kk]: Creating mo file.
[ 16%] mo-update [ko]: Creating mo file.
[ 17%] mo-update [ky]: Creating mo file.
[ 17%] mo-update [lt]: Creating mo file.
[ 17%] mo-update [lv]: Creating mo file.
[ 17%] mo-update [mr]: Creating mo file.
[ 17%] mo-update [ms]: Creating mo file.
[ 17%] Building C object lib/lua/build/CMakeFiles/lua.dir/linit.c.o
[ 18%] mo-update [nb]: Creating mo file.
[ 18%] mo-update [nl]: Creating mo file.
[ 18%] Building C object lib/lua/build/CMakeFiles/lua.dir/liolib.c.o
[ 18%] mo-update [nn]: Creating mo file.
[ 18%] mo-update [pl]: Creating mo file.
[ 18%] mo-update [pt]: Creating mo file.
[ 19%] mo-update [pt_BR]: Creating mo file.
[ 19%] mo-update [ro]: Creating mo file.
[ 19%] Building C object lib/lua/build/CMakeFiles/lua.dir/llex.c.o
[ 19%] mo-update [ru]: Creating mo file.
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/llex.c: In function ‘llex’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/llex.c:375:14: warning: this statement may fall through [-Wimplicit-fallthrough=]
  375 |         else luaX_lexerror(ls, "invalid long string delimiter", TK_STRING);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/llex.c:377:7: note: here
  377 |       case '=': {
      |       ^~~~
[ 19%] mo-update [sk]: Creating mo file.
[ 19%] mo-update [sl]: Creating mo file.
[ 19%] mo-update [sr_Cyrl]: Creating mo file.
[ 20%] mo-update [sr_Latn]: Creating mo file.
[ 20%] mo-update [sv]: Creating mo file.
[ 20%] mo-update [sw]: Creating mo file.
[ 21%] Building C object lib/lua/build/CMakeFiles/lua.dir/lmathlib.c.o
[ 21%] mo-update [tr]: Creating mo file.
[ 21%] mo-update [tt]: Creating mo file.
[ 22%] mo-update [uk]: Creating mo file.
[ 22%] Building C object lib/lua/build/CMakeFiles/lua.dir/lmem.c.o
[ 22%] mo-update [vi]: Creating mo file.
[ 22%] Building C object lib/lua/build/CMakeFiles/lua.dir/loadlib.c.o
[ 22%] mo-update [zh_CN]: Creating mo file.
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/loadlib.c: In function ‘ll_sym’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/loadlib.c:76:21: warning: ISO C forbids conversion of object pointer to function pointer type [-Wpedantic]
   76 |   lua_CFunction f = (lua_CFunction)dlsym(lib, sym);
      |                     ^
[ 22%] mo update
[ 22%] Built target translations
Scanning dependencies of target GenerateVersion
[ 22%] Built target GenerateVersion
Scanning dependencies of target IRROTHEROBJ
[ 22%] Building C object lib/lua/build/CMakeFiles/lua.dir/lobject.c.o
[ 22%] Building C object lib/lua/build/CMakeFiles/lua.dir/lopcodes.c.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/loslib.c.o
[ 23%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceSDL.cpp.o
[ 23%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceLinux.cpp.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/lparser.c.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/lstate.c.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/lstring.c.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/lstrlib.c.o
[ 23%] Building C object lib/lua/build/CMakeFiles/lua.dir/ltable.c.o
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltable.c: In function ‘luaH_get’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltable.c:473:23: warning: this statement may fall through [-Wimplicit-fallthrough=]
  473 |     case LUA_TNUMBER: {
      |                       ^
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltable.c:481:5: note: here
  481 |     default: {
      |     ^~~~~~~
[ 24%] Building C object lib/lua/build/CMakeFiles/lua.dir/ltablib.c.o
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltablib.c: In function ‘addfield’:
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltablib.c:137:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  137 |   if (!lua_isstring(L, -1))
      |   ^~
/home/xuby/Downloads/Minetest/dragonfireclient/lib/lua/src/ltablib.c:140:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  140 |     luaL_addvalue(b);
      |     ^~~~~~~~~~~~~
[ 24%] Building C object lib/lua/build/CMakeFiles/lua.dir/ltm.c.o
[ 24%] Building C object lib/lua/build/CMakeFiles/lua.dir/lundump.c.o
[ 24%] Building C object lib/lua/build/CMakeFiles/lua.dir/lvm.c.o
[ 24%] Building C object lib/lua/build/CMakeFiles/lua.dir/lzio.c.o
[ 25%] Linking C static library liblua.a
[ 25%] Built target lua
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceConsole.cpp.o
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceStub.cpp.o
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceWin32.cpp.o
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CIrrDeviceFB.cpp.o
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CLogger.cpp.o
[ 26%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/COSOperator.cpp.o
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/Irrlicht.cpp.o
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/os.cpp.o
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/leakHunter.cpp.o
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROTHEROBJ.dir/CProfiler.cpp.o
Scanning dependencies of target IRRMESHOBJ
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CSkinnedMesh.cpp.o
[ 27%] Built target IRROTHEROBJ
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CBoneSceneNode.cpp.o
[ 27%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CMeshSceneNode.cpp.o
[ 28%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CAnimatedMeshSceneNode.cpp.o
[ 28%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CB3DMeshFileLoader.cpp.o
[ 28%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/COBJMeshFileLoader.cpp.o
[ 28%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRMESHOBJ.dir/CXMeshFileLoader.cpp.o
Scanning dependencies of target IRRVIDEOOBJ
[ 28%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CVideoModeList.cpp.o
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CFPSCounter.cpp.o
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CNullDriver.cpp.o
[ 29%] Built target IRRMESHOBJ
Scanning dependencies of target IRRIOOBJ
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CFileList.cpp.o
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CFileSystem.cpp.o
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COpenGLCacheHandler.cpp.o
[ 29%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COpenGLDriver.cpp.o
[ 30%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CLimitReadFile.cpp.o
[ 30%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CMemoryFile.cpp.o
[ 30%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CReadFile.cpp.o
[ 30%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CWriteFile.cpp.o
[ 30%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CZipReader.cpp.o
[ 31%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CMountPointReader.cpp.o
[ 31%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRIOOBJ.dir/CAttributes.cpp.o
[ 31%] Built target IRRIOOBJ
Scanning dependencies of target IRROBJ
[ 31%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CBillboardSceneNode.cpp.o
[ 32%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CCameraSceneNode.cpp.o
[ 32%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COpenGLShaderMaterialRenderer.cpp.o
[ 32%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CDummyTransformationSceneNode.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COpenGLSLMaterialRenderer.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CEmptySceneNode.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CMeshManipulator.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COpenGLExtensionHandler.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLESDriver.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLESExtensionHandler.cpp.o
[ 33%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLES2Driver.cpp.o
[ 34%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLES2ExtensionHandler.cpp.o
[ 34%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLES2FixedPipelineRenderer.cpp.o
[ 34%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLES2MaterialRenderer.cpp.o
[ 34%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/COGLES2Renderer2D.cpp.o
[ 34%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CWebGL1Driver.cpp.o
[ 35%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CGLXManager.cpp.o
[ 35%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CWGLManager.cpp.o
[ 35%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CEGLManager.cpp.o
[ 35%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/mt_opengl_loader.cpp.o
[ 35%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CSceneCollisionManager.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CSceneManager.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CMeshCache.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRROBJ.dir/CDefaultSceneNodeFactory.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CColorConverter.cpp.o
[ 36%] Built target IRROBJ
Scanning dependencies of target IRRGUIOBJ
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIButton.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImage.cpp.o
[ 36%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUICheckBox.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageLoaderBMP.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageLoaderJPG.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIComboBox.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageLoaderPNG.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageLoaderTGA.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageWriterJPG.cpp.o
[ 37%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIContextMenu.cpp.o
[ 38%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRVIDEOOBJ.dir/CImageWriterPNG.cpp.o
[ 38%] Built target IRRVIDEOOBJ
[ 38%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIEditBox.cpp.o
[ 39%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIEnvironment.cpp.o
[ 39%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIFileOpenDialog.cpp.o
[ 39%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIFont.cpp.o
[ 39%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIImage.cpp.o
[ 39%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIInOutFader.cpp.o
[ 40%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIListBox.cpp.o
[ 40%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIMenu.cpp.o
[ 40%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIMeshViewer.cpp.o
[ 40%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIMessageBox.cpp.o
[ 40%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIModalScreen.cpp.o
[ 41%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIScrollBar.cpp.o
[ 41%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUISpinBox.cpp.o
[ 41%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUISkin.cpp.o
[ 41%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIStaticText.cpp.o
[ 41%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUITabControl.cpp.o
[ 42%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUITable.cpp.o
[ 42%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIToolBar.cpp.o
[ 42%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIWindow.cpp.o
[ 42%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIColorSelectDialog.cpp.o
[ 42%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CDefaultGUIElementFactory.cpp.o
[ 43%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUISpriteBank.cpp.o
[ 43%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIImageList.cpp.o
[ 43%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUITreeView.cpp.o
[ 43%] Building CXX object lib/irrlichtmt/source/Irrlicht/CMakeFiles/IRRGUIOBJ.dir/CGUIProfiler.cpp.o
[ 43%] Built target IRRGUIOBJ
Scanning dependencies of target IrrlichtMt
[ 43%] Linking CXX static library ../../lib/Linux/libIrrlichtMt.a
[ 43%] Built target IrrlichtMt
Scanning dependencies of target minetest
[ 43%] Building CXX object src/CMakeFiles/minetest.dir/client/camera.cpp.o
[ 43%] Building CXX object src/CMakeFiles/minetest.dir/client/activeobjectmgr.cpp.o
[ 43%] Building CXX object src/CMakeFiles/minetest.dir/client/client.cpp.o
[ 43%] Building CXX object src/CMakeFiles/minetest.dir/client/clientenvironment.cpp.o
In file included from /home/xuby/Downloads/Minetest/dragonfireclient/src/client/game.h:70,
                 from /home/xuby/Downloads/Minetest/dragonfireclient/src/client/gameui.h:28,
                 from /home/xuby/Downloads/Minetest/dragonfireclient/src/client/client.cpp:31:
/home/xuby/Downloads/Minetest/dragonfireclient/src/irrlicht_changes/static_text.h:188:62: error: ‘irr::io::SAttributeReadWriteOptions’ has not been declared
  188 |   virtual void serializeAttributes(io::IAttributes* out, io::SAttributeReadWriteOptions* options) const;
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xuby/Downloads/Minetest/dragonfireclient/src/irrlicht_changes/static_text.h:191:63: error: ‘irr::io::SAttributeReadWriteOptions’ has not been declared
  191 |   virtual void deserializeAttributes(io::IAttributes* in, io::SAttributeReadWriteOptions* options);
      |                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/minetest.dir/build.make:89: src/CMakeFiles/minetest.dir/client/client.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:746: src/CMakeFiles/minetest.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

--- a/README.md
+++ b/README.md
@@ -182,7 +182,7 @@ Download IrrlichtMt to `lib/irrlichtmt`, it will be used to satisfy the Irrlicht
 
     cd lib/
     wget https://github.com/minetest/irrlicht/archive/refs/tags/1.9.0mt3.tar.gz
-    tar xf master.tar.gz
+    tar xf 1.9.0mt3.tar.gz
     mv irrlicht-master irrlichtmt
     cd ..
 

--- a/README.md
+++ b/README.md
@@ -183,7 +183,7 @@ Download IrrlichtMt to `lib/irrlichtmt`, it will be used to satisfy the Irrlicht
     cd lib/
     wget https://github.com/minetest/irrlicht/archive/refs/tags/1.9.0mt3.tar.gz
     tar xf 1.9.0mt3.tar.gz
-    mv irrlicht-master irrlichtmt
+    mv irrlicht-1.9.0mt3 irrlichtmt
     cd ..
 
 Download source, without using Git:

--- a/README.md
+++ b/README.md
@@ -171,8 +171,8 @@ For Fedora users:
 
 Download source (this is the URL to the latest of source repository, which might not work at all times) using Git:
 
-    git clone --depth 1 https://github.com/EliasFleckenstein03/dragonfireclient
-    cd minetest
+    git clone --depth 1 https://github.com/dragonfireclient/dragonfireclient
+    cd dragonfireclient
 
 Download minetest_game (otherwise only the "Development Test" game is available) using Git:
 
@@ -180,7 +180,11 @@ Download minetest_game (otherwise only the "Development Test" game is available)
 
 Download IrrlichtMt to `lib/irrlichtmt`, it will be used to satisfy the IrrlichtMt dependency that way:
 
-    git clone --depth 1 https://github.com/minetest/irrlicht.git lib/irrlichtmt
+    cd lib/
+    wget https://github.com/minetest/irrlicht/archive/refs/tags/1.9.0mt3.tar.gz
+    tar xf master.tar.gz
+    mv irrlicht-master irrlichtmt
+    cd ..
 
 Download source, without using Git:
 
@@ -194,14 +198,6 @@ Download minetest_game, without using Git:
     wget https://github.com/minetest/minetest_game/archive/master.tar.gz
     tar xf master.tar.gz
     mv minetest_game-master minetest_game
-    cd ..
-
-Download IrrlichtMt, without using Git:
-
-    cd lib/
-    wget https://github.com/minetest/irrlicht/archive/master.tar.gz
-    tar xf master.tar.gz
-    mv irrlicht-master irrlichtmt
     cd ..
 
 #### Build

